### PR TITLE
Ignore injecting sidecar for none collector kind

### PR DESCRIPTION
**Pull request by hougangliu**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/747)

### CODE REVIEW

File: pkg/apis/controller/common/v1alpha3/common_types.go [hunk 1]
```diff
@@ -107,14 +107,14 @@ type FileSystemPath struct {
...
 
 const (
-	StdOutCollector           CollectorKind = "stdOutCollector"
-	FileCollector             CollectorKind = "fileCollector"
-	TfEventCollector          CollectorKind = "tfEventCollector"
-	PrometheusMetricCollector CollectorKind = "prometheusMetricCollector"
-	CustomCollector           CollectorKind = "customCollector"
+	StdOutCollector           CollectorKind = "stdout"
+	FileCollector             CollectorKind = "file"
+	TfEventCollector          CollectorKind = "tensorflow-event"
```
Comment (johnugeorge) [inline pkg/apis/controller/common/v1alpha3/common_types.go]: Won't hyphen create issues in yaml?

Comment (gaocegege) [reply pkg/apis/controller/common/v1alpha3/common_types.go]: I think it is not a key, it is a value.

/cc @hougangliu

Comment (hougangliu) [reply pkg/apis/controller/common/v1alpha3/common_types.go]: hyphen can work well in yaml
in fact, we often name a Kubernetes resource with hyphen in by yaml format

Comment (hougangliu) [reply pkg/apis/controller/common/v1alpha3/common_types.go]: yes, it's value @gaocegege

Comment (johnugeorge) [reply pkg/apis/controller/common/v1alpha3/common_types.go]: Oh, right. It is a value here.  In general, Do we have a convention to follow camel case or hyphen separated ? eg: https://github.com/kubeflow/tf-operator/blob/master/pkg/apis/common/v1/types.go#L150

Comment (hougangliu) [reply pkg/apis/controller/common/v1alpha3/common_types.go]: ok for me to use upper camel case style

**Pull request merged**